source/usr/lib/mcvirt/cluster/cluster.py

```diff
@@ -15,437 +17,555 @@
+            'password': connection_info[3],
+            'ca_cert': connection_info[4]
+        }
+        connection_info_json = json.dumps(connection_info_dict)
+        return base64.b64encode(connection_info_json)
+
+    @Pyro4.expose()
+    def print_info(self):
+        """Print information about the nodes in the cluster"""
+        table = Texttable()
+        table.set_deco(Texttable.HEADER | Texttable.VLINES)
+        table.header(('Node', 'IP Address', 'Status'))
+        # Add this node to the table
+        table.add_row((get_hostname(), self.get_cluster_ip_address(),
+                       'Local'))
+
+        # Add remote nodes
+        for node in self.get_nodes(return_all=True):
+            node_config = self.get_node_config(node)
+            node_status = 'Unreachable'
+            try:
+                self.get_remote_node(node)
+                node_status = 'Connected'
+            except CouldNotConnectToNodeException:
+                pass
+            table.add_row((node, node_config['ip_address'],
+                           node_status))
+        return table.draw()
+
+    @Pyro4.expose()
+    @locking_method()
+    def add_node_configuration(self, node_name, ip_address,
+                               connection_user, connection_password,
+                               ca_key):
+        """Add MCVirt node to configuration and generates SSH
+        authorized_keys file
+        """
+        self._get_registered_object('auth').assert_permission(PERMISSIONS.MANAGE_CLUSTER)
+
+        # Create CA file
+        ssl_object = self._get_registered_object(
+            'certificate_generator_factory').get_cert_generator(node_name)
+        ssl_object.ca_pub_file = ca_key
+
+        # Connect to node and obtain cluster user
+        remote = Connection(username=connection_user, password=connection_password,
+                            host=node_name)
+        remote_user_factory = remote.get_connection('user_factory')
+        connection_user = remote_user_factory.get_user_by_username(connection_user)
+        remote.annotate_object(connection_user)
+        username, password = connection_user.create_cluster_user(host=get_hostname())
 
-    @staticmethod
-    def getHostname():
-        """Returns the hostname of the system"""
-        return socket.gethostname()
+        # Add node to configuration file
+        def add_node_config(mcvirt_config):
+            mcvirt_config['cluster']['nodes'][node_name] = {
+                'ip_address': ip_address,
+                'username': username,
+                'password': password
+            }
+        MCVirtConfig().update_config(add_node_config)
 
-    def __init__(self, mcvirt_instance):
-        """Sets member variables"""
-        self.mcvirt_instance = mcvirt_instance
+    @Pyro4.expose()
+    @locking_method()
+    def add_node(self, node_connection_string):
+        """Connect to a remote MCVirt machine, shares SSH keys and clusters the machines."""
```

Comment:
Is this comment still accurate? It mentions SSH keys but I thought SSL is used instead now